project-chip/connectedhomeip
# Make a QueryImage command from OTA Requestor to OTA Provider

* Make ota-provider-app and ota-requestor-app compilable on Mac
* Remove self commissioning code from ota-requestor-app
* Add a parameter for IP address for OTA Requestor to use for creating secure session
* Add a parameter for fabric index for OTA Requestor to use for creating secure session
* Add a parameter for time to wait for OTA Requestor to initiate a QueryImage command from startup

#### Problem
Currently, there is no code in ota-requestor-app to open a secure session to ota-provider-app in order to send QueryImage command and begin the OTA process.

Fixes: https://github.com/project-chip/connectedhomeip/issues/9523

#### Change overview
Provided extra parameters to the ota-requestor-app to support secure session establishment and a way to trigger the QueryImage command.

After a specified amount of time (supplied by user on the command line when launching ota-requestor-app), use the `OperatioanlDeviceProxy` class to create a secure session between ota-requestor-app and ota-provider-app. Once the secure session has been established, send the `QueryImage` command to the ota-provider-app.

#### Testing
Tested manually using the ota-requestor-app and ota-provider-app:
1. Launch ota-provider-app: `chip-ota-provider-app -f test.bin`
- The `-f test.bin` is a dummy file that resides locally to ota-provider-app location
2. Commission the ota-provider-app: `chip-tool pairing onnetwork 0xBEEFFEEB 20202021`
3. Launch ota-requestor-app: `chip-ota-requestor-app -d 30 -u 5550 -i 192.168.4.55 -n 0x00000000BEEFFEEB -q 10`
- The `-d 30` is the discriminator to help with commissioning
- The `-u 5550` is the UDP port to help with commissioning
- The `-i 192.168.4.55` is the IP address of the ota-provider-app that has been resolved manually
- The `-n 0x00000000BEEFFEEB` is the node ID of the ota-provider-app
- The `-q 10` is the amount of time in seconds to wait before initiating secure session estabishment
4. Commission the ota-requestor-app: `chip-tool pairing onnetwork-long 0x1234567890 20202021 30`
5. Check that after 10 seconds of launch ota-requestor-app
- The secure session is established
- QueryImage command is received on the ota-provider-app
- QueryImageResponse is received on the ota-requestor-app
- The `test.bin` file is downloaded by the ota-requestor-app
